pkg/inputs/mysqlbatch/position_value.go

```diff
@@ -161,22 +178,77 @@ type TableStats struct {
 	Current           *TablePosition `toml:"current" json:"current"`
 	EstimatedRowCount int64          `json:"estimated-count"`
 	ScannedCount      int64          `json:"scanned-count"`
+	Done              bool           `json:"done"`
 }
 
-type BatchPositionValue struct {
-	Start       utils.MySQLBinlogPosition `toml:"start-binlog" json:"start-binlog"`
-	TableStates map[string]TableStats     `toml:"table-stats" json:"table-stats"`
+type TableStatsV1 struct {
+	Max               []TablePosition `toml:"max" json:"max"`
+	Min               []TablePosition `toml:"min" json:"min"`
+	Current           []TablePosition `toml:"current" json:"current"`
+	EstimatedRowCount int64           `json:"estimated-count"`
+	ScannedCount      int64           `json:"scanned-count"`
+	Done              bool            `json:"done"`
+}
+
+type BatchPositionValueV1Beta1 struct {
+	SchemaVersion string                    `toml:"schema-version" json:"schema-version"`
+	Start         utils.MySQLBinlogPosition `toml:"start-binlog" json:"start-binlog"`
+	TableStates   map[string]TableStats     `toml:"table-stats" json:"table-stats"`
+}
+
+type BatchPositionValueV1 struct {
+	SchemaVersion string                    `toml:"schema-version" json:"schema-version"`
+	Start         utils.MySQLBinlogPosition `toml:"start-binlog" json:"start-binlog"`
+	TableStates   map[string]TableStatsV1   `toml:"table-stats" json:"table-stats"`
 }
 
 func EncodeBatchPositionValue(v interface{}) (string, error) {
-	return myJson.MarshalToString(v)
+	v1, ok := v.(BatchPositionValueV1)
+	if !ok {
+		return "", errors.Errorf("invalid position value type: %v", reflect.TypeOf(v))
+	}
+	v1.SchemaVersion = SchemaVersionV1
+	return myJson.MarshalToString(v1)
 }
 
 func DecodeBatchPositionValue(s string) (interface{}, error) {
-	v := BatchPositionValue{}
+	v := BatchPositionValueV1{}
 	if err := myJson.UnmarshalFromString(s, &v); err != nil {
```

Comment:
I think this would fail. any test for the migration?

Reply:
made several changes, PTAL @Ryan-Git